backend-services/src/common/types.py

```diff
@@ -0,0 +1,6 @@
+from typing import NewType
+
+WalletAddress = NewType("WalletAddress", str)
```

Comment:
Is there a benefit to making a type for this over it being a string? Can we enforce, say, length? Or is this just as an example?

Reply:
Definitely, you'll especially notice this in our Rust code.   I'll sketch one example here.  Imagine a function which takes `public_key` and `secret_key` as arguments, in that order.  If these two values have different types and the caller mistakenly messes up the order of the arguments then `mypy` (Python) or the compiler (Rust) will create big red messages prompting the caller to rectify their mistake.  If both have the same type then this will go unnoticed and possibly even go into production.

Long story short, languages that support static type checking make it possible to assign types in a manner that corresponds to the semantics of the values it is intended to represent.